apps/emqx_web_hook/src/emqx_web_hook_actions.erl

```diff
@@ -335,8 +337,26 @@ method(POST) when POST == <<"POST">>; POST == <<"post">> -> post;
 method(PUT) when PUT == <<"PUT">>; PUT == <<"put">> -> put;
 method(DEL) when DEL == <<"DELETE">>; DEL == <<"delete">> -> delete.
 
-headers(Headers) ->
-    emqx_http_lib:normalise_headers(maps:to_list(Headers)).
+preproc_and_normalise_headers(Headers) ->
+    maps:fold(fun(K0, V, Acc) ->
+        case emqx_rule_utils:if_contains_placeholder(K0) of
+            false ->
+                %% see emqx_http_lib:normalise_headers/1 for more info
+                K = re:replace(K0, "_", "-", [{return, binary}]),
+                LK = string:lowercase(K),
+                Acc#{emqx_rule_utils:preproc_tmpl(LK) => emqx_rule_utils:preproc_tmpl(V)};
```

Comment:
why the preproc_tmpl for the key if it does not containe placeholder at all?

Reply:
That's because we do not know if the key contains placeholders or not at the time when we `proc_tmpl`, so we always preproc and then proc it later.

Reply:
I've changed the code